flink-runtime/src/test/java/org/apache/flink/runtime/net/SSLUtilsTest.java

```diff
@@ -438,6 +498,46 @@ public void testInvalidFingerprintParsing() throws Exception {
 
     // ------------------------------- utils ----------------------------------
 
+    private static class TestSslContextSupplier implements Supplier<SslContext> {
+        private static final ThreadLocal<Integer> callCount = ThreadLocal.withInitial(() -> 0);
```

Comment:
It's so we can check if the supplier was actually used.  Since its created inside `SSLUtils` there's no other good way I can think of to get the state of the instance that was created (and make sure it actually was created).

Reply:
I was just being safe in case multiple tests run in parallel, the thread-local keeps them from sharing the other's state.

Reply:
If it's multi-threaded then why not AtomicInteger if it's not multi-threaded then why not dropping `ThreadLocal`. Maybe there is a reason but now I don't see.

Reply:
> if it's not multi-threaded
It could be depending on how the test runner depends to run the tests though.  This is just being safe and making sure that the tests don't share static state if they happen to run in parallel in the same JVM.

If you really feel strongly about it I can remove the ThreadLocal and just use a static variable, I'm just trying to be safe and not write flaky tests.